client/src/views/common/BoostView.tsx

```diff
@@ -32,61 +41,86 @@ function onDeleteConfig(configType: ConfigT, name: string) {
   console.log(`name: ${name}`);
 }
 
-// Only dummy data
-const baseConfigs: BoostConfig[] = [
-  {
-    type: 'powerPlan',
-    options: [
-      { fileName: 'my_plan_1.json', displayName: 'my_plan_1' },
-      { fileName: 'this_one_gets_you_to_144.json', displayName: '144' },
-    ],
-    active: { fileName: 'my_plan_1.json', displayName: 'my_plan_1' },
-  },
-  {
-    type: 'rider',
-    options: [
-      { fileName: 'al.json', displayName: 'AL' },
-      { fileName: 'charles_rider.json', displayName: 'charles' },
-    ],
-    active: { fileName: 'charles_rider.json', displayName: 'charles' },
-  },
-  {
-    type: 'bike',
-    options: [
-      { fileName: 'blacksmith.json', displayName: 'Black Smith' },
-      { fileName: 'wombat.json', displayName: 'Wombat' },
-      { fileName: 'precilla.json', displayName: 'Precilla' },
-    ],
-    active: { fileName: 'wombat.json', displayName: 'Wombat' },
-  },
-  {
-    type: 'track',
-    options: [
-      { fileName: 'ford.json', displayName: 'Ford' },
-      { fileName: 'holden_track.json', displayName: 'Holden' },
-      { fileName: 'battle_mountain.json', displayName: 'Battle Mountain' },
-    ],
-    active: { fileName: 'ford.json', displayName: 'Ford' },
-  },
-];
-
 /**
  * Boost View component
  *
  * @returns {React.Component} Component
  */
 export default function BoostView() {
   // TODO: remove the hardcoded value for `distTravelled` with actual value read from MQTT
+  const [dist, setDist] = useState(0);
+  const [distOffset, setDistOffset] = useState<number | null>(null);
+  const [configs, setConfigs] = useState<BoostConfig[]>([
+    {
+      type: 'powerPlan',
+      options: [],
+    },
+    {
+      type: 'rider',
+      options: [],
+    },
+    {
+      type: 'bike',
+      options: [],
+    },
+    {
+      type: 'track',
+      options: [],
+    },
+  ]);
+
+  // fetch the reed distance from wireless module #3
+  const reedDistance = useSensorData(3, Sensor.ReedDistance, Number);
+
+  useEffect(() => {
+    if (reedDistance) {
+      // Set distance to 2 decimal places
+      setDist(Math.round(reedDistance * 100) / 100);
+    }
+  }, [reedDistance]);
+
+  const handleConfigsReceived = useCallback(
+    (configsReceived: Static<typeof ConfigPayloadRT>) => {
+      // Replace the available config options with those received
+      setConfigs(
+        configs.map((config) => {
+          return { ...config, options: configsReceived[config.type] };
+        }),
+      );
+    },
+    [configs],
+  );
+
+  const handleDistOffsetReceived = useCallback(
+    (payload: Static<typeof RecommendedSPRT>) => {
+      setDistOffset(payload.distanceOffset);
+    },
+    [],
+  );
+
+  useChannelShaped('boost/configs', ConfigPayloadRT, handleConfigsReceived);
+  useChannelShaped(
+    'boost/recommended_sp',
+    RecommendedSPRT,
+    handleDistOffsetReceived,
+  );
+
+  const handleReset = () => {
+    setDistOffset(0);
+    toast.success('Calibration Reset!');
+    resetCalibration();
```

Comment:
Small nitpick, I am of the opinion that success messages should only be sent after the action has been completed successfully. For example, if there was an exception during the execution of `resetCalibration()`, then the success message could be misleading.

Reply:
Agreed, I don't even know why I had it before the `resetCalibration()` call.